lib/ddtrace/auto_instrument.rb

```diff
@@ -0,0 +1,41 @@
+require 'ddtrace'
+
+module Datadog
+  # AutoInstrumentation enables all integrations
+  module AutoInstrument
+    module_function
+
+    def patch_all
+      integrations = []
+      Datadog.registry.each do |integration|
+        # some instrumentations are automatically enabled when the `rails` instrumentation is enabled,
+        # patching them on their own automatically outside of the rails integration context would
+        # cause undesirable service naming, so we exclude them based their auto_instrument? setting.
+        # we also don't want to mix rspec/cucumber integration in as rspec is env we run tests in.
+        next unless integration.klass.auto_instrument?
+        integrations << integration.name
+      end
+
+      Datadog.configure do |c|
+        c.reduce_log_verbosity
+        # This will activate auto-instrumentation for Rails
+        integrations.each do |integration_name|
+          c.use integration_name
+        end
+      end
+    end
+
+    if Datadog::Contrib::Rails::Utils.railtie_supported?
```

Comment:
I don't think this is a good idea. There's a deliberate design decision to keep integrations (Rails) separate from the core (`AutoInstrument`), and this creates coupling between the two. This means we always have to load Rails to do any kind of Ruby instrumentation, something we want to avoid if we want to move contributions eventually to their own separate package.

I would suggest instead implementing some kind of callback pattern: have the auto-instrumentation code here emit an  event in which the loaded Rails integration can hook into, receive, and act accordingly. It may not be totally that simple, but its definitely worthwhile avoiding this coupling.